bin_heap.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+
+
+class BinaryHeap(object):
+    def __init__(self):
+        self.heaplist = [0]
+        self.currentsize = 0
+
+    def push(self, val):
+        self.heaplist.append(val)
+        self.currentsize += 1
+        self.bubble_up(self.currentsize)
```

Comment:
this is definitely the right way to do this.  A helper method works great for this sort of thing.